torch/_dynamo/symbolic_convert.py

```diff
@@ -670,7 +670,11 @@ def LOAD_DEREF(self, inst):
         self.push(self.symbolic_locals[inst.argval])
 
     def STORE_FAST(self, inst):
-        self.symbolic_locals[inst.argval] = self.pop()
+        val = self.pop()
+        if hasattr(val, "mutable_local") and val.mutable_local:
+            if not hasattr(val.mutable_local, "source"):
+                val.mutable_local.source = LocalSource(inst.argval)
+        self.symbolic_locals[inst.argval] = val
```

Comment:
This doesn't look right.  Side effects are applied before locals are updated, and none of this should be needed for something that gets `.copy()`ed

Reply:
If you .copy() a ListVariable, the new ListVariable's mutable_local does not have a source. Is there a better place to bind it?

Reply:
Or, we should remove the mutable_local requirement for append to work...

Reply:
Anything not an input should have `source=None`.  `val.mutable_local.source` should not be used in this example.